тестовое

MaksSemkin opened the pull request.

MaksSemkin committed:
тестовое

idbolshakov reviewed (changes requested):
привет, спасибо за выполнение ТЗ. К решению есть несколько вопросов.

1. Если на странице открыть попап с товаром несколько раз, то при нажатии на крестик (закрытие попапа) он не закрывается с первого раза

idbolshakov commented on src/index.css:
```diff
@@ -0,0 +1,195 @@
+
+button {
+
+    /*display: block;*/
```
зачем оставил это в решении?

MaksSemkin replied:
исправил

idbolshakov commented on src/index.css:
```diff
@@ -0,0 +1,195 @@
+.wrap {
```
знакомы ли какие-то системы организации css кода? Почему решил сложить все стили в один файл?

idbolshakov commented on src/index.js:
```diff
@@ -0,0 +1,104 @@
+    const el = document.createElement('div');
+    el.classList ="wrapCart";
+    el.innerHTML = ` 
+            <button class="close" onClick = "closeCard()" style="position: absolute; right: 0px" >X</button>
```
инлайновое определение обработчиков событий - плохая практика

idbolshakov commented on src/index.js:
```diff
@@ -0,0 +1,104 @@
+
+} )
+
+let inOrderButtons = document.querySelectorAll('.inOrder');
```
почему тут let  а не const или var ?

MaksSemkin replied:
исправил

idbolshakov commented on src/index.js:
```diff
@@ -0,0 +1,104 @@
+ }
+
+API.products.forEach(({id, img, price, title})=> {
+       const el = document.createElement('div')
```
тут используешь `createElement` ниже `inerHTML` почему бы и этот `div `  в `innerHTML` не засунуть?

idbolshakov commented on src/index.js:
```diff
@@ -0,0 +1,104 @@
+    <button class = "inCart">В корзину</button>
+</div>`
+    let closeButtons = document.querySelectorAll('.close');
+    closeButtons.forEach((item)=>{item.addEventListener('click',closeCard)})
```
для 10 товаров у тебя будет 10 обработчиков, для 100 - 100 и тд. Есть идеи как оптимизировать?

MaksSemkin replied:
Идея есть. Кинуть один обработчик на страницу, создать условие по таргету и уже ,исходя из условия, показывать окно корзины или заказа.

MaksSemkin committed:
исправление замечаний

1.Корзина и Заказ закрываются с первого раза.
2.Убрал неактивный код
3.Оптимизировал обработчики.
